### PR TITLE
Add go-structurizr

### PR DESCRIPTION
**Pull request by krzysztofreczek**
This PR adds automatically generated C4 diagrams to all golang services using https://github.com/krzysztofreczek/go-structurizr library.

### CODE REVIEW

File: internal/trainer/service/application.go [hunk 1]
```diff
@@ -13,10 +13,10 @@ import (
 )
 
 func NewApplication(ctx context.Context) app.Application {
-	firestoreClient, err := firestore.NewClient(ctx, os.Getenv("GCP_PROJECT"))
-	if err != nil {
-		panic(err)
-	}
+	firestoreClient := &firestore.Client{}
```
Comment (krzysztofreczek) [inline internal/trainer/service/application.go:16]: @roblaszczak, I am wondering how to initialize the Firestore client when generating C4 diagrams. I have a couple of ideas:
1. Generate diagrams in docker where Firestore emulator is available
2. Do simple DI with an auxiliary mocked configuration

What would you recommend?

Comment (roblaszczak) [reply internal/trainer/service/application.go:16]: ad 1. it doesn't need to be generated **in** Docker, but **with** Docker, right? I guess that passing envs and running docker with Firestore is fine and should work pretty out of the box :wink:

File: internal/c4/go.mod [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module github.com/ThreeDotsLabs/wild-workouts-go-ddd-example/internal/c4
```
Comment (m110) [inline internal/c4/go.mod:1]: Thanks for contributing!

Perhaps we could create a top-level `tools` directory and keep `c4` there? To not confuse people looking for services or application code in `internal`. 🙂

File: internal/trainer/service/application.go (hunk 1 above)
Comment (krzysztofreczek) [reply internal/trainer/service/application.go]: Apparently, after I faked a couple of env vars it worked without docker-compose running

File: internal/c4/go.mod (hunk 2 above)
Comment (krzysztofreczek) [reply internal/c4/go.mod]: https://github.com/ThreeDotsLabs/wild-workouts-go-ddd-example/pull/36/commits/ea2032e34a0adfeb62f79c1b550e7be2fb8a4cba

**Pull request merged**